daffodil/parser.py

```diff
@@ -138,6 +138,9 @@ def condition(self):
         self.value()
         self.consume_whitespace(newlines=False)
 
+        if self.tokens[-2].content == "?=" and not isinstance(self.tokens[-1], Boolean):
+            raise ValueError('"?=" operator requires boolean value (true/false)')
```

Comment:
This is where I moved the validation about `?=` only allowing boolean values in the expression